src/main.cpp

```diff
@@ -2295,82 +2252,124 @@ bool Session::InitializePeriod(bool isSessionOpened /* = false */)
       return false;
   }
 
-  while ((adp = adaptiveTree_->GetAdaptationSet(i++)))
+  uint32_t adpIndex{0};
+  adaptive::AdaptiveTree::AdaptationSet* adp{nullptr};
+
+  while ((adp = adaptiveTree_->GetAdaptationSet(adpIndex++)))
   {
     if (adp->representations_.empty())
       continue;
 
-    bool manual_streams = adp->type_ == adaptive::AdaptiveTree::StreamType::VIDEO
-                              ? manual_streams_ != 0
-                              : manual_streams_ == 1;
+    bool isManualStreamSelection;
+    if (adp->type_ == adaptive::AdaptiveTree::StreamType::VIDEO)
+      isManualStreamSelection = m_settingStreamSelection != SETTINGS::StreamSelection::AUTO;
+    else
+      isManualStreamSelection = m_settingStreamSelection == SETTINGS::StreamSelection::MANUAL;
 
-    // Select good video stream
-    adaptive::AdaptiveTree::Representation* defaultRepresentation =
-        adaptiveTree_->ChooseRepresentation(adp);
-    size_t repId = manual_streams ? adp->representations_.size() : 0;
+    // Get the default initial stream repr. based on "adaptive repr. chooser"
+    adaptive::AdaptiveTree::Representation* defaultRepr{adaptiveTree_->ChooseRepresentation(adp)};
 
-    do
+    if (isManualStreamSelection)
     {
-      m_streams.push_back(std::make_unique<STREAM>(
-          *adaptiveTree_, adp, media_headers_, representationChooser_,
-          m_kodiProps.m_playTimeshiftBuffer, repId, first_period_initialized_));
-      STREAM& stream(*m_streams.back());
+      // Add all stream representations
+      for (size_t reprIdx{0}; reprIdx < adp->representations_.size(); reprIdx++)
```

Comment:
this `reprIdx` can be a little confusing to read next to `reprIndex`, maybe just sticking with the usual `i` would make for cleaner reading and less chance to accidentally use the wrong one?

Reply:
restored as `i`